### ISSUE TITLE
Incorrect rendering of services.picom.settings in configuration.nix man page

### ISSUE THREAD

**Issue by xaverdh** [NixOS/nixpkgs#86402]
**Describe the bug**
The `Type` of  `services.picom.settings` in the `configuration.nix` man page is:

> list of boolean or signed integer or floating point number or string or list of boolean or signed integer or floating point number or string or list of boolean or signed integer or floating point number or strings or attribute set of boolean or signed integer or floating point number or stringss or attribute set of boolean or signed integer or floating point number or string or list of boolean or signed integer or floating point number or strings or attribute set of boolean or signed integer or floating point number or stringsss or attribute set of boolean or signed integer or floating point number or string or list of boolean or signed integer or floating point number or string or list of boolean or signed integer or floating point number or strings or attribute set of boolean or signed integer or floating point number or stringss or attribute set of boolean or signed integer or floating point number or string or list of boolean or signed integer or floating point number or strings or attribute set of boolean or signed integer or floating point number or stringsss

Apart from this being totally crazy, note that `stringsss` should probably be `strings`.

**Notify maintainers**
cc @rnhmjoj

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
 - system: `"x86_64-linux"`
 - host os: `Linux 5.6.7, NixOS, 20.09pre223385.3b00e78e63b (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `"nixos-20.09pre223385.3b00e78e63b"`
 - channels(pantarhei): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixos modules affected by the problem
module: picom
```

**Comment by xaverdh** [comment]
Apart from the `stringsss` issue, the `Type` rendering would probably benefit from brackets in nested types..

**Comment by rnhmjoj** [comment]
Thank you for opening an issue, I did notice this a while ago but I had forgotten about it.

The type is actually correct: these `settings` options are inherently complex because they map a subset of Nix to a configuration format (JSON, INI, etc.). When you unfold their type (which is what the manual shows) it can get really complicated or even infinite if the type is recursive. 
The problem is the readily solved manually adding a meaningful description instead of letting ~Gollum~ Nix generate one.

This brings us to the second issue: the "sss" come from the implementation of the `listOf` type description, which is something like this
```
listOf = type: {
  ...
  description = "list of ${type.description}s"
}
```
This works as inteded for the simple case of a "primitive" type like `listOf int`, which gives "a list of ints" but produces `list of list of list ofs intsss` when nested multiple times.
As you said, this could be addressed by moving away from the current natural language kind of description to something more structured, using brackets. I'm not so sure about this, though.

Currently I'm working on a cleanup of the picom module, when done the description would read

![picom.settings](https://user-images.githubusercontent.com/2817565/80870183-2109bc00-8ca5-11ea-969a-6e6ee2ecdcde.png)

**Comment by xaverdh** [comment]
That does look a lot better :-)
Thanks for the work!

**Comment by rnhmjoj** [comment]
So, I have fixed Picom but this is a general problem, unfortunately.
I have found two other options with a more or less crazy type:

- `services.clight.settings`

- `services.pdns-recursor.settings` (my bad)

and about a dozen options with "ss" by looking for `Type: .+ of \w+ss` in `configuration.nix(5)`.

**Comment by xaverdh** [comment]
I am closing this, because the concrete issue with `picom` is fixed, but something more general like #88828 is still very much appreciated I think!
